test/test_document_symbols.jl

```diff
@@ -0,0 +1,156 @@
+module test_document_symbols
+
+using Test
+using JETLS
+using JETLS: JL, JS
+using JETLS: DocumentSymbol, SymbolKind, Position, get_toplevel_symbols!, get_symbols!
+
+get_toplevel_symbols(ex::JL.SyntaxTree) = get_toplevel_symbols!(ex, DocumentSymbol[])
+get_symbols(ex::JL.SyntaxTree) = get_symbols!(ex, DocumentSymbol[])
+
+function test_symbol(s::DocumentSymbol,
+                     name::String,
+                     kind::SymbolKind.Ty,
+                     start_pos,
+                     end_pos,
+                     children_len)
+    @test s.name == name
+    @test s.kind == kind
+    @test s.range.start == Position(; line = start_pos[1], character = start_pos[2])
+    @test s.range.var"end" == Position(; line = end_pos[1], character = end_pos[2])
+    if children_len == 0
+        @test isnothing(s.children) || isempty(s.children)
+    else
+        @test !isnothing(s.children)
+        @test length(s.children) == children_len
+    end
+end
+
+"""
+The `end` keyword's final byte is offset by -1. Until that's fixed, use this function to
+test document symbols whose spans end with an `end`.
+"""
+function test_symbol_end(s::DocumentSymbol,
+                         name::String,
+                         kind::SymbolKind.Ty,
+                         start_pos,
+                         end_pos,
+                         children_len)
+    end_pos = (end_pos[1], end_pos[2] - 1)
+    return test_symbol(s, name, kind, start_pos, end_pos, children_len)
+end
+
+@testset "Simple expressions" begin
+    src = """
```

Comment:
```suggestion
    let src = """
```
I'd prefer wrapping each independent test cases within `let` block so that we don't have unexpected reuse of bindings from previous test cases.